CarboPlanner/src/views/HomeScreen.tsx

```diff
@@ -49,70 +50,78 @@ export default function HomeScreen(){
     }
 
-    const renderMealItem = ({ item }: { item: jsonData }) => {
-      const uuidKey = Object.keys(item)[0]; // Extracting the UUID key
-      const uuidObject = item[uuidKey]; // Getting the object with the UUID key
-      const mealsArray = uuidObject.meals; // Accessing the meals array from the UUID object
+    const renderMealItem = ({ item }) => {
+      // deconstruct item into its key value pair 
+      const [uuidKey, meal] = item; 
+
+    
+      const mealsArray = meal.meals // Access the meals array from the UUID object
+    
       return (
-        <List name={uuidObject.name} imageURI={uuidObject.icon} uuidKey={uuidKey}>
+        <List key={uuidKey} name={meal.name} imageURI={meal.icon} uuidKey={uuidKey}>
           <HorizontalLine />
-          {mealsArray.map((meal, index) => (
-            <ListItem
-              key={index.toString()} // Ensure each item has a unique key
-              weight={meal.weight}
-              name={meal.name}
-              calories={meal.calories}
-              protein={meal.protein}
-              carbs="32" // Assuming the carbs value is fixed
-              fat={meal.fat}
-            />
-          ))}
+          {mealsArray.map((mealData, index) => {
+            const [mealUuid, mealInfo] = Object.entries(mealData)[0]; // Extract the UUID and meal info
+            return (
+              <ListItem
+                key={`${uuidKey}_${index}`} // Ensure each item has a unique key
+                weight={mealInfo.weight}
+                name={mealInfo.name}
+                calories={mealInfo.calories}
+                protein={mealInfo.protein}
+                carbs={mealInfo.carbs}
+                fat={mealInfo.fat}
+              />
+            );
+          })}
         </List>
       );
     };
+    
+    
+    
+    
+
 
   return (
     <View style={styles.container}>
-      <View style={{flexDirection: 'row', alignItems: 'center'}}>
-        <Text style={styles.header_text}>Today</Text>
-        <TouchableOpacity onPress={goToCalendar}>
-          <View style={styles.calendarIconContainer}>
-            <Image source={require("../../assets/func-icon/calendar.png")} style={styles.calendarIcon} resizeMode="contain" />
-          </View>
-        </TouchableOpacity>
-      </View>
+    <View style={{flexDirection: 'row', alignItems: 'center'}}>
+      <Text style={styles.header_text}>Today</Text>
+      <TouchableOpacity onPress={goToCalendar}>
+        <View style={styles.calendarIconContainer}>
+          <Image source={require("../../assets/func-icon/calendar.png")} style={styles.calendarIcon} resizeMode="contain" />
+        </View>
+      </TouchableOpacity>
+    </View>
 
       <Card>
         <View style={styles.row}>
           <CircularSlider value={1200} max={2000}/>
         </View>
         <View style={styles.row}>
-            <MacroProgressBar name={"Carbs"} value={50} max={210} />
+            <MacroProgressBar name={"Carbs"} value={0.5} max={210} />
```

Comment:
Changed in another branch
```suggestion
            <MacroProgressBar name={"Carbs"} value={50} max={210} />
```